diff --git a/package.json b/package.json
index e36b3a9..ed1a8ed 100644
--- a/package.json
+++ b/package.json
@@ -53,6 +53,7 @@
     "@jupyterlab/filebrowser": "^3.0.11",
     "@jupyterlab/launcher": "^3.0.9",
     "@jupyterlab/launcher-extension": "^3.0.11",
+    "@jupyterlab/mainmenu": "^3.0.11",
     "@jupyterlab/services": "^6.0.9",
     "@jupyterlab/translation": "^3.0.9",
     "@jupyterlab/ui-components": "^3.0.7",
diff --git a/src/index.ts b/src/index.ts
index e44ffb2..719070a 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -5,6 +5,8 @@ import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin
 } from '@jupyterlab/application';
+import { IMainMenu } from '@jupyterlab/mainmenu';
+// import { Menu } from '@lumino/widgets';
 
 import { ICommandPalette } from '@jupyterlab/apputils';
 import { swanProjectIcon } from './icons';
@@ -32,17 +34,19 @@ import { kernelsInfoRequest } from './request';
 const extension: JupyterFrontEndPlugin<void> = {
   id: 'swanprojects',
   autoStart: true,
-  optional: [ILauncher],
-  requires: [ICommandPalette],
+  optional: [],
+  requires: [ICommandPalette,IMainMenu,ILauncher],
   activate: async (
     app: JupyterFrontEnd,
     palette: ICommandPalette,
-    launcher: ILauncher | null,
+    launcher: ILauncher,
     labShell: ILabShell,
+    mainMenu: IMainMenu,
     browserFactory: IFileBrowserFactory
   ) => {
     console.log('JupyterLab extension swanprojects is activated!');
     const { commands } = app;
+    
     commands.addCommand(CommandIDs.projectDialog, {
       icon: swanProjectIcon,
       label: 'New',
@@ -104,6 +108,22 @@ const extension: JupyterFrontEndPlugin<void> = {
         category: PALETTE_CATEGORY
       });
     }
+
+    // // Add the command to the main menu
+    if (mainMenu){
+    // mainMenu.addMenu(tutorialMenu, { rank: 80 });
+
+    // Add the command to the menu
+      // const menu = new Menu({ commands });
+      // menu.title.label = 'New Project';
+      // mainMenu.addMenu(menu, { rank: 80 });
+      // menu.addItem({
+      // command: CommandIDs.projectDialog,
+      // args: {},
+      // });
+
+      //mainMenu.fileMenu.addItem({"name":"New Project",})
+    }
   }
 };
 
